akka-stream/src/main/scala/akka/stream/impl/fusing/ActorInterpreter.scala

```diff
@@ -191,12 +226,9 @@ private[akka] class ActorOutputBoundary(val actor: ActorRef, debugLogging: Boole
 
   override def onPush(elem: Any, ctx: BoundaryContext): Directive = {
     onNext(elem)
-    if (downstreamDemand > 0) ctx.pull()
+    if (downstreamDemand > 0) tryBounceBall(ctx)
     else if (downstreamCompleted) ctx.finish()
```

Comment:
swap these, check for completion first. This is cancellation and it is out-of-band.